### PR TITLE
[CHAT-1477] partial update channels 

### PR DESCRIPTION
**Pull request by thesyncim**
Partial Update
A partial update can be used to set and unset specific fields when it is necessary to retain additional custom data fields on the object. AKA a patch style update.

```javascript
// Here's a channel with some custom field data that might be useful
channel = client.channel(type, id , {
   source: "user",
   source_detail:{ user_id: 123 },
   channel_detail:{ topic: "Plants and Animals", rating: "pg" }
})

//  let's change the source of this channel
await channel.updatePartial({ set:{ source: "system" } });

//  since it's system generated we no lonfer need source_detail
await channel.updatePartial({ unset: ["source_detail"] });

// set nested field
await channel.updatePartial({ set: { "channel_detail.topic": "Nature" } });

// remove the topic property from channel_detail object 
await channel.updatePartial({ unset: ["channel_detail.topic"] });
```

### CODE REVIEW

**Comment by vishalnarkhede** [comment]
Build will be good after this - https://github.com/GetStream/stream-chat-js/pull/551/files

File: src/channel.ts [hunk 1]
```diff
@@ -525,6 +527,21 @@ export class Channel<
...
 
+  /**
+   * updatePartial - partial update channel properties
+   *
+   * @param {PartialChannelUpdate<ChannelType>} partial update request
+   *
+   * @return {Promise<PartialUpdateChannelAPIResponse<ChannelType,CommandType, UserType>>}
+   */
+  async updatePartial(update: PartialChannelUpdate<ChannelType>) {
+    return await this.getClient().patch<
```
Comment (mahboubii) [inline src/channel.ts:538]:
```suggestion
    return this.getClient().patch<
```

File: src/channel.ts [hunk 2]
```diff
@@ -525,6 +527,21 @@ export class Channel<
...
+      PartialUpdateChannelAPIResponse<ChannelType, CommandType, UserType>
+    >(this._channelURL(), {
+      ...update,
+    });
```
Comment (mahboubii) [inline src/channel.ts:542]:
```suggestion
    >(this._channelURL(), update);
```

File: src/channel.ts [hunk 3]
```diff
@@ -27,6 +27,8 @@ import {
   MessageFilters,
   MuteChannelAPIResponse,
   PaginationOptions,
+  PartialChannelUpdate,
```
Comment (mahboubii) [inline src/channel.ts:30]: maybe rename this to `PartialUpdateChannel`  to be more consistent

**Review by AnatolyRugalev — APPROVED** [review]
LGTM

File: src/channel.ts (hunk 1 above)
Comment (ferhatelmas) [reply src/channel.ts:538]: @thesyncim why did you resolve ?

**Pull request merged**